internal/querycoordv2/meta/channel_dist_manager.go

```diff
@@ -374,19 +388,27 @@ func (m *ChannelDistManager) GetShardLeader(channelName string, replica *Replica
 					case !candidatesServiceable && channelServiceable:
 						// Current candidate is not serviceable but new channel is
 						updateNeeded = true
+						logger.Debug("set updateNeeded in case 1")
 					case candidatesServiceable == channelServiceable && channel.Version > candidates.Version:
 						// Same service status but higher version
 						updateNeeded = true
+						logger.Debug("set updateNeeded in case 2",
+							zap.Int64("candidates version", candidates.Version),
+							zap.Int64("channel version", channel.Version))
 					}
-
+					logger.Info("check updateNeeded", zap.Bool("updateNeeded", updateNeeded))
```

Comment:
prefer `set serviceable delegator with larger version to candidate shard leader` and add `zap.String("node", channel.Node)`